internal/controller/controllers/infraenv_controller.go

```diff
@@ -794,7 +825,9 @@ func (r *InfraEnvReconciler) handleEnsureISOErrors(
 			}
 		}
 	}
-
+	if !r.ImageServiceEnabled {
```

Comment:
Splitting the code from the status update might also help here since this is currently never called if the image service is not enabled.

It would be good to just not have this check here at all and somehow make this a part of that new function that is only ever used when the image service is enabled.